dotCMS/src/main/java/com/dotmarketing/portlets/osgi/AJAX/OSGIBaseAJAX.java

```diff
@@ -77,6 +85,30 @@ public void service(HttpServletRequest request, HttpServletResponse response) th
 
 	}
 
+	/**
+	 * Extracted so it can be mocked
+	 * @param user
+	 * @return
+	 * @throws DotDataException
+	 */
+	@VisibleForTesting
+	boolean doesUserHaveAccessToPortlet(final User user) throws DotDataException {
+	  return APILocator.getLayoutAPI().doesUserHaveAccessToPortlet("dynamic-plugins", user);
+	}
+
+	/**
+	 * extracted as a separated method, so it can be verified when called
+	 * @param method
+	 * @param parameterTypes
+	 * @return
+	 * @throws NoSuchMethodException
+	 */
+	@VisibleForTesting
+	Method getMethod(String method, Class<?>... parameterTypes ) throws NoSuchMethodException {
```

Comment:
We have a ReflectionUtils, may be better there

Reply:
First this is a deprecated class Im not spending too much effort on it..  and the only reason I extracted that logic into a separate method was to verify that the method gets called using mokito @jdotcms